tests/SoftDeletesTest.php

```diff
@@ -62,4 +62,31 @@ public function it_can_include_only_trashed()
             $query
         );
     }
+
+    /** @test */
+    public function it_can_disable_soft_deletes_when_using_an_alias()
+    {
+        // making sure the query doesn't fail
+        UserProfile::query()
+            ->joinRelationship('user', function ($join) {
+                $join->withTrashed();
```

Comment:
Was your intent to call `->as()` here too?

Reply:
I guess that would make more sense, to keep the test in line with the tother tests.